src/hashset.c

```diff
@@ -1429,16 +1423,16 @@ int hashsetNext(hashsetIterator *iter, void **elemptr) {
     while (1) {
         if (iter->index == -1 && iter->table == 0) {
             /* It's the first call to next. */
+            if (iter->hashset->tables[0] == NULL) {
+                /* empty hashset, we're done */
+                break;
+            }
```

Comment:
Doesn't this break the fingerprint check in `hashsetResetIterator` for safe iterators? The fingerprint has not been initialized yet here.

Do we have a test case for iterating over an empty table using safe iterator?

Reply:
That's true! I'll add a UT.